File: graphGenerationCodePy/Bipartite/BipartiteGraph.py
import os
import random

def get_string():
    return input()

def get_int():
    return int(input())

def get_real():
    return float(input())

def main():
    print("\n\n---------------------------------------------------")
    n = int(input("Enter number of nodes on the source side: \t"))
    m = int(input("Enter number of nodes on the sink side: \t"))
    max_probability = float(input("Enter max probability: \t\t\t\t"))
    if max_probability > 1:
        print("Max probability should be less than or equal to 1")
        return
    min_capacity = int(input("Enter minimum capacity: \t\t\t"))
    max_capacity = int(input("Enter maximum capacity: \t\t\t"))
    directory = os.getcwd()
    file_name = input("Enter the output file name: \t\t\t") + ".txt"
    print("---------------------------------------------------\n")

    try:
        with open(os.path.join(directory, file_name), 'w') as outfile:
            edge = [[0 for _ in range(m)] for _ in range(n)]
            for i in range(n):
                for j in range(m):
                    value = random.random()
                    if value <= max_probability:
                        edge[i][j] = value
                    else:
                        edge[i][j] = 0

            print("-----------------------------------------")
            print("\tSource\tSink\tCapacity")
            print("-----------------------------------------") 

            for i in range(n):
                value = min_capacity + int(random.random() * (max_capacity - min_capacity + 1))
                print("\ts\tl" + str(i + 1) + "\t" + str(value))
                outfile.write("\ts\tl" + str(i + 1) + "\t" + str(value) + "\n")

            for i in range(n):
                for j in range(m):
                    if edge[i][j] > 0:
                        edge[i][j] = min_capacity + int(edge[i][j] * (max_capacity - min_capacity + 1))
                        print("\tl" + str(i+1) + "\tr" + str(j+1) + "\t" + str(int(edge[i][j])))
                        outfile.write("\tl" + str(i+1) + "\tr" + str(j+1) + "\t" + str(int(edge[i][j])) + "\n")

            for j in range(m):
                value = min_capacity + int(random.random() * (max_capacity - min_capacity + 1))
                print("\tr" + str(j+1) + "\tt\t" + str(value))
                outfile.write("\tr" + str(j + 1) + "\tt\t" + str(value) + "\n")

            print("\n\nOutput is created at: \t" + os.path.join(directory, file_name))

    except Exception as ex:
        print(ex)

if __name__ == "__main__":
    main()

File: graphGenerationCode/testBi/BipartiteGraph.java
import java.io.*;

public class BipartiteGraph {
    public static void main(String[] args) throws Exception {
        int maxCapacity, minCapacity = 1;
        int n, m = 50;
        double maxProbability = 0.4;

        for (maxCapacity = 1; maxCapacity <= 101; maxCapacity+=10) {
            // int n = 50;  // Number of nodes at end equals number of nodes at start
 
            n = 50;
            m = 50;

            // Construct file name
            String fileName = "bi-" + n + "-" + m + "-" + maxProbability + "-" + minCapacity + "-" + maxCapacity + ".txt";

            // Generate graph and write to file
            generateAndWriteGraph(n, m, maxProbability, minCapacity, maxCapacity, fileName);
        }
    }

    // Method for graph generation and file writing
    private static void generateAndWriteGraph(int n, int m, double maxProbability, int minCapacity, int maxCapacity, String fileName) throws IOException {
        String directory = System.getProperty("user.dir");
        // Append the subdirectory 'bi-m\' to the directory path
        File subDirectory = new File(directory, "bi-c");
        if (!subDirectory.exists()) {
            subDirectory.mkdirs();  // Create the subdirectory if it does not exist
        }

        File file = new File(subDirectory, fileName);
        PrintWriter outFile = new PrintWriter(new FileWriter(file));

        double[][] edge = new double[n][m];
        double value, x;
        System.out.println("\n\n---------------------------------------------------");
        System.out.println("Creating file: " + file.getPath());
        System.out.println("---------------------------------------------------\n");

        // Generating the graph
        for (int i = 0; i < n; i++) {
            for (int j = 0; j < m; j++) {
                value = Math.random();
                if (value <= maxProbability)
                    edge[i][j] = value;
                else
                    edge[i][j] = 0;
            }
        }

        // Printing and writing the edges
        System.out.println("-----------------------------------------");
        System.out.println("\tSource\tSink\tCapacity");
        System.out.println("-----------------------------------------");

        // Computing the edges out of source
        for (int i = 0; i < n; i++) {
            x = Math.random();
            value = Math.floor(minCapacity + (x * (maxCapacity - minCapacity + 1)));
            System.out.println("\ts" + "\tl" + (i + 1) + "\t" + (int) value);
            outFile.println("s" + "\tl" + (i + 1) + "\t" + (int) value);
        }

        // Computing for the vertices between source and sink
        for (int i = 0; i < n; i++) {
            for (int j = 0; j < m; j++) {
                if (edge[i][j] > 0) {
                    edge[i][j] = Math.floor(minCapacity + (edge[i][j] * (maxCapacity - minCapacity + 1)));
                    System.out.println("\tl" + (i + 1) + "\tr" + (j + 1) + "\t" + (int) edge[i][j]);
                    outFile.println("l" + (i + 1) + "\tr" + (j + 1) + "\t" + (int) edge[i][j]);
                }
            }
        }

        // Computing the edges into the sink
        for (int j = 0; j < m; j++) {
            x = Math.random();
            value = Math.floor(minCapacity + (x * (maxCapacity - minCapacity + 1)));
            System.out.println("\tr" + (j + 1) + "\tt" + "\t" + (int) value);
            outFile.println("r" + (j + 1) + "\tt" + "\t" + (int) value);
        }

        System.out.println("\n\nOutput is created at: \t" + directory + "\\" + fileName);
        outFile.close();
    }
}
